src/renderer/services/midgard/shares.ts

```diff
@@ -37,14 +43,36 @@ const createSharesService = (
    */
   const shares$ = (address: Address): PoolSharesLD =>
     FP.pipe(
-      Rx.combineLatest([api$, reloadShares$]),
-      RxOp.map(([api]) => api),
+      Rx.combineLatest([api$, reloadSharesWithTimer$]),
+      RxOp.switchMap(([api, delayTime]) =>
```

Comment:
`delayTime = 0` in case you wan't to support `observableState<number | undefined>(0)`

Reply:
actually this is a thing that i wanted to support: to have [reloadShares](https://github.com/thorchain/asgardex-electron/pull/1153/files#diff-0ddf52bae472e9088bc56e84ec464a0c5b090d17e60db08e03b35675e0ac8f6bR154) as a function with non-required parameter to use it just as a single function everywhere where no need to have a delay. for example for the manual [reload button](https://github.com/thorchain/asgardex-electron/blob/develop/src/renderer/views/deposit/DepositView.tsx#L103)

`Rx.timer` has a `0` value inside by default 
![image](https://user-images.githubusercontent.com/67793194/111869886-7cfb3300-8992-11eb-90d0-81862b5e2da5.png)

anyway i've moved [default value for the reloadShares](https://github.com/thorchain/asgardex-electron/pull/1153/commits/818fa6c4ced52178617ead69b143d6f85efdce29)

Reply:
> actually this is a thing that i wanted to support: to have reloadShares as a function with non-required parameter to use it just as a single function everywhere where no need to have a delay.

Make sense! So why have you changed it with last commit to `observableState<number>(0)`?

> Rx.timer has a 0 value inside by default

True. However, by reading your code you can't know if `Rx.timer` excepts `undefined` or not. That's why `Rx.timer(undefined)` looks weird at first view and there is nothing wrong to give extra information using `delayTime = 0` then. Anyway, just a nitpick...

Reply:
>Make sense! So why have you changed it with 818fa6c to observableState<number>(0)?
just to make things clearer as you suggested and pass a `0` value explicitly with
```    reloadShares: (delayTime = 0) => reloadSharesWithTimer(delayTime),```